### PR TITLE
Docs QA updates

### LINKED ISSUE #815: merge readme-dev into CONTRIBUTING.md
### Description

README-dev.md doesn't appear to be linked from anywhere and contains mostly the same information as CONTRIBUTING.md has or should have. I propose we flesh out CONTRIBUTING.md and remove README-dev.md.

### Success criteria (What criteria will consider this ticket closeable?)
CONTRIBUTING.md should walk a new contributor through contributing to Pilosa from installing Go onward. I think it should also include a note that the same process (as far as git/github goes) applies to all repos in our organization, so that we can link to it from other repositories.

### PR DESCRIPTION
**Pull request by alanbernstein**
## Overview

A few simple docs fixes and additions, including:

- external tutorials section
- readme for a useful landing page for visitors to github.com/pilosa/pilosa/docs
- bunch of glossary fixes
- formatting and link fixes
- fixes #815 
- pilosa/general#29

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Pilosa docs are maintained here, to stay in sync with the codebase.
+
+Please visit [our website](https://www.pilosa.com/docs) to view the docs complete with styles, diagrams, and comprehensive search.
```
Comment (codysoyland) [inline docs/README.md]: Always end pilosa.com urls with a slash, as they always get redirected to include a slash.

File: docs/examples.md [hunk 2]
```diff
@@ -263,7 +263,7 @@ python import_from_sdf.py -p <path_to_sdf_file> -file id_fingerprint.csv
 ```
 
 
-First, follow the instruction in the [getting started]({{< ref "getting-started.md" >}}) guide to run a Pilosa server. Then create the indexes and frames according to the schemas outlined in the Data Model section above.
+First, follow the instruction in the [getting started](../getting-started.md) guide to run a Pilosa server. Then create the indexes and frames according to the schemas outlined in the Data Model section above.
```
Comment (codysoyland) [inline docs/examples.md]: Fix link (delete `.md`). There are several more examples below.

File: docs/glossary.md [hunk 3]
```diff
@@ -6,53 +6,62 @@ nav = []
...
 
-<strong id="anti-entropy">Anti-entropy:</strong> A periodic process that compares each slice and its replicas across the cluster to repair inconsistencies.
+<strong id="pql">[PQL](../query-language/):</strong> Pilosa Query Language.
 
-<strong id="node">Node:</strong> An individual running instance of Pilosa server which belongs to a cluster.  
+<strong id="protobuf">[Protobuf](https://developers.google.com/protocol-buffers/):</strong> Protocol Buffers is a binary serialization format which Pilosa uses for internal messages, and can be used by clients as an alternative to JSON.
 
-<strong id="cluster">Cluster:</strong> A cluster consists of one or more nodes which share a cluster configuration. The cluster also defines how data is replicated throughout and how internode communication is coordinated. Pilosa does not have a leader node, all data is evenly distributed, and any node can respond to queries.
+<strong id="range">[Range](../query-lanuage/#range):</strong>: A [PQL](#pql) query that returns bits based on comparison to timestamps, set according to the [time quantum](#time-quantum).
```
Comment (codysoyland) [inline docs/glossary.md]: Spelling query-lanuage -> query-language -- another instance below.

**Comment by alanbernstein** [comment]
Yeah, I've found a few broken external links using https://www.deadlinkchecker.com, I'll fix those. 

I'm also working on checking internal anchor links, I don't expect many problems right now but as docs evolve I would like to be able to find problems easily.

**Comment by codysoyland** [comment]
This is mostly looking good, but could you see about changing all the internal links to include a slash at the end? That would avoid a redirect (giving us referrers in analytics and lower latency) and I think the anchors get wiped during the redirect.

**Comment by codysoyland** [comment]
All the latest changes are great!

File: docs/webui.md [hunk 4]
```diff
@@ -30,7 +32,7 @@ In addition to standard PQL, the console supports a few special commands, prefix
 - `:create frame <framename>`
 - `:delete frame <framename>`
 
-Frame creation also supports options like `timeQuantum` or `inverseEnabled`. When creating a new frame, add options by using the keys documented in [API reference](../api-reference).
+Index and frame creation also supports options like timeQuantum or inverseEnabled. Check the API reference sections for [index](../api-reference/#change-index-time-quantum) and [frame](../api-reference/#create-frame) options to see other available keys.
```
Comment (alanbernstein) [inline docs/webui.md]: this link will work after merging with master

**Comment by alanbernstein** [comment]
@codysoyland I found and fixed a couple more small issues when I checked the mac/linux install sections. I'm done now, but I'll wait for you to look at those if you want

File: docs/tutorials.md [hunk 5]
```diff
@@ -8,6 +8,12 @@ nav = [
 ]
 +++
 
+### External Tutorials
```
Comment (codysoyland) [inline docs/tutorials.md]: As we discussed IRL, this would be nice as a Note admonition just below the Tutorials heading.

**Pull request merged**